src/App.tsx

```diff
@@ -15,36 +15,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = 'default',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+const preparedGoods = [...goodsFromServer];
+
+const sortGoods = (order: SortType, isReversed: boolean): string[] => {
+  let sortedGoods: string[];
+
+  switch (order) {
+    case SortType.ALPHABET:
+      sortedGoods = [...preparedGoods].sort((a, b) => a.localeCompare(b));
+      break;
+    case SortType.LENGTH:
+      sortedGoods = [...preparedGoods].sort((a, b) => a.length - b.length);
+      break;
+    default:
+      sortedGoods = [...preparedGoods];
+  }
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(preparedGoods);
```

Comment:
you don't need another state fo the data here it's redundant. Your page reloads based on click, not data